serde-processor/src/main/java/io/micronaut/serde/processor/SerdeAnnotationVisitor.java

```diff
@@ -687,6 +686,16 @@ private void visitSubtype(ClassElement supertype, ClassElement subtype, VisitorC
             return;
         }
 
+        if (!subtype.hasAnnotation(SerdeConfig.SerIgnored.class)) {
+            // Replicate the Jackson behaviour of using the ignore annotation from supertype but allow to override it
+            AnnotationValue<SerdeConfig.SerIgnored> serIgnored = supertype.getAnnotation(SerdeConfig.SerIgnored.class);
```

Comment:
what if it's on the supersupertype

Reply:
The annotation is not inherited

Reply:
in a hierarchy of A extends B and B extends C, and C has @SerIgnored, I think this will mean that A will only get SerIgnored if B is visited before A?

Reply:
No, `@SerIgnored` is not inherited, so adding it to `B` shouldn't add it to `A`.

Reply:
but youre copying it here? i dont understand what youre saying. this code copies the annotation from the supertype, but not if it's further up, even though i assume in jackson it would still be considered for such a hierarchy

Reply:
Well if the annotation is not inherited, putting it on B shouldn’t be present on A etc.

Reply:
but with this code, putting it on B does copy it to A.

Reply:
The subtypes in this code are the subtypes of the base class defined by JsonSubTypes. The logic is called one. It’s not called on every type.

Reply:
See https://github.com/micronaut-projects/micronaut-serialization/pull/640/files#diff-37b66da15700d594aac6c8e6eab4763e00f8dc3b0d769186e203b400a7a557b7R800

The ignore is not inherited. With subtypes it acts the same, if you put ignore again and want to ignore something from the supertype you need to repeat it.

Reply:
ive pushed a pendingfeature with the setup that i mean. the test passes with jackson-databind.

Reply:
Ahh, looks like Jackson respects the last definition. I can take a look in the next PR